internal/providers/pluginfw/resources/cluster/data_cluster.go

```diff
@@ -0,0 +1,106 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/compute_tf"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func DataSourceCluster() datasource.DataSource {
+	return &ClusterDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &ClusterDataSource{}
+
+type ClusterDataSource struct {
+	Client *common.DatabricksClient
+}
+
+type ClusterInfo struct {
+	ClusterId   types.String               `tfsdk:"cluster_id" tf:"optional,computed"`
+	Name        types.String               `tfsdk:"cluster_name" tf:"optional,computed"`
+	ClusterInfo *compute_tf.ClusterDetails `tfsdk:"cluster_info" tf:"optional,computed"`
+}
+
+func (d *ClusterDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "databricks_cluster_pluginframework"
```

Comment:
One thing I realized: the prefix `databricks` should be supplied in the `req` field, so you can do
```go
resp.TypeName = req.ProviderTypeName + "_cluster_pluginframework"
```

I realize that the implementation of `Metadata()` for the provider is probably incorrect. It is currently `databricks-tf-provider`. However, the docs for the TypeName are:
```
	// TypeName should be the provider type. For example, examplecloud, if
	// the intended resource or data source types are examplecloud_thing, etc.
```
This probably needs to be changed to `databricks`.

Reply:
Thanks for the review, sounds good, since this needs to be done for quality monitor and volumes as well, I will do this in a separate PR to isolate changes.